### PR TITLE
Feat/add roles

### PR DESCRIPTION
**Pull request by Gurr1**

### CODE REVIEW

File: backend/src/main/java/it/chalmers/gamma/controller/administrationSubControllers/AuthorityAdminController.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.constraints.Null;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@RestController
+@RequestMapping("/admin/authorization")
+public class AuthorityAdminController {
```
Comment (Portals) [inline backend/src/main/java/it/chalmers/gamma/controller/administrationSubControllers/AuthorityAdminController.java]: Should be named `AuthorizationAdminController` to match the request mapping

File: backend/src/main/java/it/chalmers/gamma/controller/administrationSubControllers/AuthorityAdminController.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+                                    MembershipService membershipService){
+        this.authorityService = authorityService;
+        this.fkitService = fkitService;
+        this.postService = postService;
+        this.authorityLevelService = authorityLevelService;
+        this.membershipService = membershipService;
+    }
+
+    @RequestMapping(method = RequestMethod.POST)
+    public ResponseEntity<String> addAuthority(@RequestBody AuthorizationRequest request){      // TODO CHECK IF EXISTS
```
Comment (Portals) [inline backend/src/main/java/it/chalmers/gamma/controller/administrationSubControllers/AuthorityAdminController.java]: TODO CHECK IF EXISTS

File: backend/src/main/java/it/chalmers/gamma/controller/administrationSubControllers/AuthorityAdminController.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        return new AuthorityRemovedResponse();
+    }
+
+    @RequestMapping(method = RequestMethod.GET)
+    public ResponseEntity<List<Authority>> getAllAuthorities(){
+        List<Authority> authorities = authorityService.getAllAuthorities();
+        return new GetAllAuthoritiesResponse(authorities);
+    }
+
+    // BELOW THIS SHOULD MAYBE BE MOVED TO A DIFFERENT FILE
```
Comment (Portals) [inline backend/src/main/java/it/chalmers/gamma/controller/administrationSubControllers/AuthorityAdminController.java]: // BELOW THIS SHOULD MAYBE BE MOVED TO A DIFFERENT FILE

File: backend/src/main/java/it/chalmers/gamma/controller/administrationSubControllers/UserAdminController.java [hunk 4]
```diff
@@ -92,7 +92,7 @@ private UserAdminController(ITUserService itUserService, UserWebsiteService user
         return new UserCreatedResponse();
     }
     @RequestMapping(value = "/reset_password", method = RequestMethod.POST)
-    public ResponseEntity<String> resetPasswordRequest(@RequestBody ResetPasswordRequest request){
+    public ResponseEntity<String> resetPasswordRequest(@RequestBody ResetPasswordRequest request){      //TODO MOVE THIS TO ITUSERCONTROLLER
```
Comment (Portals) [inline backend/src/main/java/it/chalmers/gamma/controller/administrationSubControllers/UserAdminController.java]: TODO

File: backend/src/main/java/it/chalmers/gamma/controller/administrationSubControllers/UserAdminController.java [hunk 5]
```diff
@@ -112,7 +112,7 @@ private UserAdminController(ITUserService itUserService, UserWebsiteService user
         return new PasswordResetResponse();
     }
     @RequestMapping(value = "/reset_password/finish", method = RequestMethod.PUT)
-    public ResponseEntity<String> resetPassword(@RequestBody ResetPasswordFinishRequest request){
+    public ResponseEntity<String> resetPassword(@RequestBody ResetPasswordFinishRequest request){     //TODO MOVE THIS TO ITUSERCONTROLLER
```
Comment (Portals) [inline backend/src/main/java/it/chalmers/gamma/controller/administrationSubControllers/UserAdminController.java]: TODO

File: backend/src/main/java/it/chalmers/gamma/db/entity/Authority.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Authority authority = (Authority) o;
+        return Objects.equals(id, authority.id) &&
+                Objects.equals(authorityLevel, authority.authorityLevel);
+    }
+
+    @Override
+    public int hashCode() {
+
```
Comment (Portals) [inline backend/src/main/java/it/chalmers/gamma/db/entity/Authority.java]: Remove

File: backend/src/main/java/it/chalmers/gamma/db/entity/ITUser.java [hunk 7]
```diff
@@ -265,20 +274,21 @@ public String toString() {
         return "ITUser{" +
                 "id=" + id +
                 ", cid='" + cid + '\'' +
-                ", pass='<redacted>'" +
+                ", password='" + password + '\'' +
```
Comment (Portals) [inline backend/src/main/java/it/chalmers/gamma/db/entity/ITUser.java]: This should be `<redacted>`

File: backend/src/main/java/it/chalmers/gamma/db/entity/pk/AuthorityPK.java [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        AuthorityPK that = (AuthorityPK) o;
+        return Objects.equals(fkitGroup, that.fkitGroup) &&
+                Objects.equals(post, that.post);
+    }
+
+    @Override
+    public int hashCode() {
+
```
Comment (Portals) [inline backend/src/main/java/it/chalmers/gamma/db/entity/pk/AuthorityPK.java]: Remove

File: backend/src/main/java/it/chalmers/gamma/requests/AuthorizationRequest.java [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+        if (o == null || getClass() != o.getClass()) return false;
+        AuthorizationRequest that = (AuthorizationRequest) o;
+        return Objects.equals(post, that.post) &&
+                Objects.equals(group, that.group) &&
+                Objects.equals(authority, that.authority);
+    }
+
+    @Override
+    public int hashCode() {
+
```
Comment (Portals) [inline backend/src/main/java/it/chalmers/gamma/requests/AuthorizationRequest.java]: Remove

File: backend/src/main/resources/db/migration/db.sql [hunk 10]
```diff
@@ -1,7 +1,7 @@
 create table text (
   id  uuid constraint text_pk primary key,
   sv  text not null,
-  en  text
+  es  text
```
Comment (Portals) [inline backend/src/main/resources/db/migration/db.sql]:
```suggestion
  en text
```

File: backend/src/main/java/it/chalmers/gamma/controller/administrationSubControllers/AuthorityAdminController.java (hunk 1 above)
Comment (Gurr1) [reply backend/src/main/java/it/chalmers/gamma/controller/administrationSubControllers/AuthorityAdminController.java]: No

File: backend/src/main/java/it/chalmers/gamma/controller/administrationSubControllers/AuthorityAdminController.java (hunk 3 above)
Comment (Gurr1) [reply backend/src/main/java/it/chalmers/gamma/controller/administrationSubControllers/AuthorityAdminController.java]: This is meant in a final build with more functions

File: backend/src/main/java/it/chalmers/gamma/controller/administrationSubControllers/AuthorityAdminController.java (hunk 1 above)
Comment (Portals) [reply backend/src/main/java/it/chalmers/gamma/controller/administrationSubControllers/AuthorityAdminController.java]: Can we call it either the one or the other

Comment (Gurr1) [reply backend/src/main/java/it/chalmers/gamma/controller/administrationSubControllers/AuthorityAdminController.java]: I'd rather change the endpoint than the class name. Makes more sense internally

Comment (Portals) [reply backend/src/main/java/it/chalmers/gamma/controller/administrationSubControllers/AuthorityAdminController.java]: Do it 👍

File: backend/src/main/java/it/chalmers/gamma/controller/administrationSubControllers/UserAdminController.java (hunk 5 above)
Comment (Gurr1) [reply backend/src/main/java/it/chalmers/gamma/controller/administrationSubControllers/UserAdminController.java]: Not a part of this feature branch

**Pull request merged**